### PR TITLE
Error for an invalid cache operation while in passive mode added

### PR DESCRIPTION
**Pull request by kmajzero**
Signed-off-by: Krzysztof Majzerowicz-Jaszcz <krzysztof.majzerowicz-jaszcz@intel.com>

### CODE REVIEW

**Comment by robertbaldyga** [comment]
add to whitelist

File: inc/ocf_err.h [hunk 1]
```diff
@@ -134,6 +134,9 @@ typedef enum {
 
 	/** Cache initialized with wrong cache line size */
 	OCF_ERR_CACHE_LINE_SIZE_MISMATCH,
+
+	/** Invalid operation for cache in passive state. */
+	OCF_ERR_INVALID_OPERATION_CACHE_PASSIVE,
```
Comment (robertbaldyga) [inline inc/ocf_err.h:139]: This is rather lengthy name. How about just `OCF_ERR_CACHE_IS_PASSIVE`? We already have very similarly constructed names, like `OCF_ERR_CACHE_EXIST`.

**Review by arutk — DISMISSED** [review]

File: inc/ocf_err.h [hunk 2]
```diff
@@ -136,7 +136,7 @@ typedef enum {
 	OCF_ERR_CACHE_LINE_SIZE_MISMATCH,
 
 	/** Invalid operation for cache in passive state. */
-	OCF_ERR_INVALID_OPERATION_CACHE_PASSIVE,
+	OCF_ERR_CACHE_PASSIVE,
```
Comment (mmichal10) [inline inc/ocf_err.h:139]: Please fixup this commit with the previous one

File: src/mngt/ocf_mngt_core.c [hunk 3]
```diff
@@ -1081,7 +1086,10 @@ int ocf_mngt_core_set_seq_cutoff_promotion_count_all(ocf_cache_t cache,
 {
 	OCF_CHECK_NULL(cache);
 
-	return ocf_core_visit(cache,
+	if (ocf_cache_is_passive(cache))
+		return -OCF_ERR_CACHE_PASSIVE;
+	else
```
Comment (mmichal10) [inline src/mngt/ocf_mngt_core.c:1091]: The `else` closure is redundant

File: inc/ocf_mngt.h [hunk 4]
```diff
@@ -859,10 +859,12 @@ int ocf_mngt_cache_promotion_set_policy(ocf_cache_t cache, ocf_promotion_t type)
  * @brief Get promotion policy in given cache
  *
  * @param[in] cache Cache handle
+ * @param[out] type Policy type
  *
- * @retval Currently set promotion policy type
+ * @retval 0 success
+ * @retval Non-zero Error occurred and policy type has not been set
```
Comment (rafalste) [inline inc/ocf_mngt.h:865]: "policy type could not be retrieved" as it is a _get_ function

File: src/mngt/ocf_mngt_cache.c [hunk 5]
```diff
@@ -2918,6 +2918,9 @@ int ocf_mngt_cache_set_mode(ocf_cache_t cache, ocf_cache_mode_t mode)
 
 	OCF_CHECK_NULL(cache);
 
+	if (ocf_cache_is_passive(cache))
```
Comment (rafalste) [inline src/mngt/ocf_mngt_cache.c:2921]: ocf_cache_is_standby()

File: src/mngt/ocf_mngt_cache.c [hunk 6]
```diff
@@ -2949,24 +2955,31 @@ int ocf_mngt_cache_promotion_set_policy(ocf_cache_t cache, ocf_promotion_t type)
...
 
 int ocf_mngt_cache_promotion_get_param(ocf_cache_t cache, ocf_promotion_t type,
 		uint8_t param_id, uint32_t *param_value)
 {
 	int result;
 
+	if (ocf_cache_is_passive(cache))
+		return -OCF_ERR_CACHE_PASSIVE;
+
+
```
Comment (rafalste) [inline src/mngt/ocf_mngt_cache.c:2982]: double new line

File: inc/ocf_err.h [hunk 7]
```diff
@@ -134,6 +134,9 @@ typedef enum {
 
 	/** Cache initialized with wrong cache line size */
 	OCF_ERR_CACHE_LINE_SIZE_MISMATCH,
+
+	/** Invalid operation for cache in passive state. */
+	OCF_ERR_CACHE_PASSIVE,
```
Comment (karolinavelkaja) [inline inc/ocf_err.h:139]: passive -> standby :)

**Review by rafalste — DISMISSED** [review]

File: src/mngt/ocf_mngt_cache.c [hunk 8]
```diff
@@ -2804,7 +2804,7 @@ static void _ocf_mngt_cache_standby_complete(ocf_cache_t cache, void *priv1,
 	if (error)
 		OCF_CMPL_RET(cache, priv2, error);
 
-	_ocf_mngt_cache_set_passive(cache);
+	_ocf_mngt_cache_set_standby(cache);
 	ocf_cache_log(cache, log_info, "Successfully binded\n");
```
Comment (Deixx) [inline src/mngt/ocf_mngt_cache.c:2808]: binded -> bound

**Review by Deixx — DISMISSED** [review]

**Review by rafalste — DISMISSED** [review]

**Review by arutk — DISMISSED** [review]

**Review by rafalste — APPROVED** [review]

**Review by arutk — APPROVED** [review]

**Pull request merged**